src/features/Classes/EnrollStudent/index.jsx

```diff
@@ -0,0 +1,158 @@
+    const formData = new FormData(form);
+
+    const { studentEmail } = Object.fromEntries(formData.entries());
+
+    formData.delete('studentEmail');
+    formData.append('identifiers', studentEmail);
+    formData.append('action', 'enroll');
+    formData.append('auto_enroll', 'true');
+    formData.append('email_students', 'true');
+
+    try {
+      setLoading(true);
+
+      const response = await handleEnrollments(formData, classId);
+      const validationEmailList = response?.data?.results;
+      const messages = await getMessages();
+      let textToast = '';
+
+      /**
+       * This is because the service that checks the enrollment status is a different
+       * endpoint, and that endpoint always returns a status 200, so the error cannot be
+       * caught with a .catch.
+       */
+      if (messages?.data?.results[0]?.tags === 'error') {
+        setToastMessage(decodeURIComponent(messages?.data?.results[0]?.message));
+        setShowToast(true);
+
+        return onClose();
+      }
+
+      textToast = emailValidationMessages(validationEmailList);
+      setToastMessage(textToast);
+
+      const params = {
+        class_name: className,
+        limit: true,
+        page: initialPage,
+      };
+
+      dispatch(fetchStudentsData(username, params));
+
+      dispatch(fetchAllClassesData(username, { class_id: classId }));
+
+      setShowToast(true);
+      return onClose();
+    } catch (error) {
+      return logError(error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  return (
+    <>
+      <Toast
+        onClose={() => setShowToast(false)}
+        show={showToast}
+        className="toast-message"
+        data-testid="toast-message"
+      >
+        {toastMessage}
+      </Toast>
+      <ModalDialog
+        title="Invite student to enroll"
+        isOpen={isOpen}
+        onClose={onClose}
+        hasCloseButton
+        className="enroll-modal"
+      >
+        <ModalDialog.Header>
+          <ModalDialog.Title>Invite student to enroll</ModalDialog.Title>
+        </ModalDialog.Header>
+        <ModalDialog.Body className="body-container h-100">
+          <p className="text-uppercase font-weight-bold sub-title">
+            Class: {className}
+          </p>
+          {isLoading && (
+            <div className="w-100 h-100 d-flex justify-content-center align-items-center">
+              <Spinner
+                animation="border"
+                className="mie-3"
+                screenReaderText="loading"
+              />
+            </div>
+          )}
+          {!isLoading && (
+            <Form onSubmit={handleEnrollStudent}>
+              <FormGroup controlId="studentInfo">
+                <Form.Control
+                  as="textarea"
+                  placeholder="Enter email of the student to enroll"
+                  floatingLabel="Email"
+                  className="my-4 mr-0 student-email"
+                  name="studentEmail"
+                  required
```

Comment:
I could avoid this restriction just with a single `.` I suggest to verify if the email is an valid email before make the request

![image](https://github.com/user-attachments/assets/6c79860c-cbe3-4814-aaed-3f8e13bb36ee)

Reply:
As I said in the previous comment this api accepts also usernames, also in frontend we can't cover all the cases to say if is an email valid that the reason backend is in charge of the validation.